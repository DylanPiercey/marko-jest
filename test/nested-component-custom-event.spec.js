const path = require('path');
const { initComponent, createTestSandbox } = require('../test-utils');

describe('component with nested components', () => {
  const componentClass = initComponent(path.resolve(__dirname, './resources/click-and-win/index.marko'));
  const MODEL = {
    items: [
      { label: 'click me', value: 0 },
      { label: 'click me', value: 0 },
      { label: 'click me', value: 0 },
      { label: 'click me', value: 'win' },
      { label: 'click me', value: 0 }
    ]
  };

  let testSandbox;
  let component;

  beforeEach(() => {
    testSandbox = createTestSandbox();
  });

  afterEach(() => {
    testSandbox.reset();
  });

  describe('given default model', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, MODEL);
    });

    it('should render correctly', () => {
      expect(testSandbox.getRenderedNodes()).toMatchSnapshot();
    });

    it('should able to get nested element with getEl()', () => {
      const el = component.getEl('title');
      expect(el).toBeDefined();
    });

    it('should able to get nested components with getComponents()', () => {
      const buttons = component.getComponents('secret-item');
      expect(buttons).toHaveLength(MODEL.items.length);
    });

    it('should able to get a nested component with getComponent()', () => {
      const video = component.getComponent('award-video');
      expect(video).toBeDefined();
    });
  });
});
