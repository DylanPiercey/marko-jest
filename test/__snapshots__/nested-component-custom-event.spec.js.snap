// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component with nested components given default model should render correctly 1`] = `
Array [
  <h1>
    Click any button below and win an ipod!
  </h1>,
  <p />,
  <ul>
    <li>
      click me
    </li>
    <li>
      click me
    </li>
    <li>
      click me
    </li>
    <li>
      click me
    </li>
    <li>
      click me
    </li>
  </ul>,
]
`;
